tools/flight_recorder/components/config_manager.py

```diff
@@ -19,6 +19,7 @@ def __init__(self: "JobConfig"):
         )
         self.parser.add_argument(
             "trace_dir",
+            nargs="?",
```

Comment:
I added an assert for this instead because child class cannot override it which makes it required internally. And we don't want that behavior.